spec/services/symphony_client_spec.rb

```diff
@@ -145,4 +145,135 @@
+                      "holdType": 'TITLE',
+                      "holdRange": 'SYSTEM',
+                      "fillByDate": '2021-03-17' })
+        .to_return(status: 500, body: { message: 'User already has a hold on this material' }.to_json)
+
+      stub_request(:post, uri)
+        .with(body: { "itemBarcode": 'no_date_item_barcode',
+                      "patronBarcode": '1234',
+                      "pickupLibrary": {
+                        "resource": '/policy/library',
+                        "key": 'UP-PAT'
+                      },
+                      "holdType": 'TITLE',
+                      "holdRange": 'SYSTEM' })
+        .to_return(status: 200, body: { key: 'other_hold_key' }.to_json)
+    end
+
+    let(:uri) { "#{Settings.symws.url}/circulation/holdRecord/placeHold" }
+    let(:user) do
+      User.new(username: 'zzz123',
+               name: 'Zeke',
+               patron_key: 'some_patron_key',
+               session_token: 'e0b5e1a3e86a399112b9eb893daeacfd')
+    end
+
+    let(:patron) { instance_double(Patron, barcode: '1234', library: 'UP-PAT') }
+    let(:hold_args) { { pickup_library: 'UP-PAT', pickup_by_date: '2021-03-17' } }
+
+    context 'when place hold is successful' do
+      let(:item_barcode) { 'success_item_barcode' }
+
+      it 'returns the hold key' do
+        place_hold_response = client.place_hold(patron, user.session_token, item_barcode, hold_args)
+
+        expect(JSON.parse(place_hold_response)).to include 'key' => 'some_hold_key'
+      end
+    end
+
+    context 'when place hold fails' do
+      let(:item_barcode) { 'fail_item_barcode' }
+
+      it 'returns the reason as the error message' do
+        place_hold_response = client.place_hold(patron, user.session_token, item_barcode, hold_args)
+
+        expect(JSON.parse(place_hold_response)).to include 'message' => 'User already has a hold on this material'
+      end
+    end
+
+    context 'when place hold does not include fill by date' do
+      let(:item_barcode) { 'no_date_item_barcode' }
+
+      it 'returns the hold key' do
+        hold_args = { pickup_library: 'UP-PAT' }
+        place_hold_response = client.place_hold(patron, user.session_token, item_barcode, hold_args)
+
+        expect(JSON.parse(place_hold_response)).to include 'key' => 'other_hold_key'
+      end
+    end
+  end
+
+  describe '#get_hold_info' do
+    before do
+      stub_request(:get, uri)
+        .with(query: hash_including(includeFields: include_fields))
+        .to_return(status: 200, body: { resource: '/circulation/holdRecord' }.to_json)
+    end
+
+    let(:hold_key) { 'a_hold_key' }
+    let(:uri) { "#{Settings.symws.url}/circulation/holdRecord/key/#{hold_key}" }
+    let(:include_fields) { '*,item{*,bib{title,author},call{*}}' }
+
+    let(:user) do
+      User.new(username: 'zzz123',
+               name: 'Zeke',
+               patron_key: 'some_patron_key',
+               session_token: 'e0b5e1a3e86a399112b9eb893daeacfd')
+    end
```

Comment:
Can you move one of these `let(:user)` statements up to the top of the file (under `let(:client)`)? And delete all the other let user statements? Will kill about 20-30 lines

Reply:
sure, let me try, there might be one spec for unauthenticated user, I will add clear the user for that one.

Reply:
you don't need to, `let` lazy loads...

Reply:
commit coming in a bit :)